File: day-4/part-1.rs
use std::fs::File;
use std::io::{self, prelude::*, BufReader};

fn main() -> io::Result<()> {
    let file = File::open("day-4/input")?;
    let reader = BufReader::new(file);

    let mut current: String = "".to_string();
    let mut valid = 0;

    for line in reader.lines() {
        let line_str: String = line?;
        if !(line_str.eq("")) {
            current.push_str(&line_str);
        } else {
            if
                current.contains("byr:") &&
                current.contains("iyr:") &&
                current.contains("eyr:") &&
                current.contains("hgt:") &&
                current.contains("hcl:") &&
                current.contains("ecl:") &&
                current.contains("pid:")
            {
                valid = valid + 1;
            }
            current = "".to_string();
        }
    }
    println!("{}", valid);

    Ok(())
}

File: day-4/part-1.py
import os
import re
dirname = os.path.dirname(__file__)
filename = os.path.join(dirname, 'input')

input = open(filename, "r")

current = ""

lines = input.readlines()
last = lines[-1]
valid = 0

for line in lines:
    if line == last:
        current = line

    if line == "\n" or line == last:
        regex = r".*(?=.*pid:)(?=.*byr:)(?=.*iyr:)(?=.*eyr:)(?=.*hgt:)(?=.*hcl:)(?=.*ecl:).*"
        if re.match(regex, current):
            valid += 1
        current = ""
    else:
        current += line.strip()

print(valid)
